packages/realm-react/src/cachedCollection.ts

```diff
@@ -158,7 +158,10 @@ export function createCachedCollection<T extends Realm.Object>({
   };
 
   if (!isDerived) {
-    cachedCollectionResult.addListener(listenerCallback);
+    // Add this on the next tick, in case there is a write transaction occuring immediately after creation of this collection
+    setImmediate(() => {
```

Comment:
I wonder if [queueMicrotask](https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask) might be appropriate. My understanding is it fires slightly earlier than `setImmediate`/`setTimeout`, not sure if it would be "too early" or not

Reply:
Seems to function in both cases. I'll have to review both methods to decide which is better suited for this.

Reply:
Ahh OK, let's leave it as is then!